kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriberServiceImpl.java

```diff
@@ -461,20 +472,27 @@ public ChildImportRejection updateChildSubscriber(Long msisdn, MctsChild childUp
                         if (subscriberByMctsId.getMother() == null) {
                             subscriberByMctsId.setMother(childUpdate.getMother());
                         }
-                        finalSubscription = updateOrCreateSubscription(subscriberByMctsId, subscription, dob, pack, language, circle, SubscriptionOrigin.MCTS_IMPORT);
+                        finalSubscription = updateOrCreateSubscription(subscriberByMctsId, subscription, dob, pack, language, circle, SubscriptionOrigin.MCTS_IMPORT, false);
                     } else {
                         //A different subscriber found with same mobile number
                         isSameSubscriber = false;
                     }
                 }
-                if(!isSameSubscriber) {
+                if (!isSameSubscriber) {
                     return childRejectionMcts(convertMapToChild(record), false, RejectionReasons.MOBILE_NUMBER_ALREADY_SUBSCRIBED.toString(), action);
                 }
             }
         }
 
+        liveBirthChildDeathCheck(finalSubscription, record);
+
+        return childRejectionMcts(convertMapToChild(record), true, null, action);
+
+    }
+
+    private void liveBirthChildDeathCheck(Subscription subscription, Map<String, Object> record) {
         // a child subscription was created -- deactivate mother's pregnancy subscription if she has one
-        Subscriber subscriber = finalSubscription != null ? finalSubscription.getSubscriber() : null;
+        Subscriber subscriber = subscription != null ? subscription.getSubscriber() : null;
         Subscription pregnancySubscription = subscriptionService.getActiveSubscription(subscriber,
```

Comment:
we should call this only when subscriber != null because based on above stmt, subscriber could be null !!